mlos_bench/mlos_bench/optimizers/mlos_core_optimizer.py

```diff
@@ -127,11 +126,12 @@ def bulk_register(self,
 
     def _adjust_signs_df(self, df_scores: pd.DataFrame) -> pd.DataFrame:
         """
-        In-place adjust the signs of the scores for MINIMIZATION problem.
+        Coerce optimization target scores to floats
+        and adjust the signs for MINIMIZATION problem.
         """
-        for (opt_target, opt_dir) in self._opt_targets.items():
-            df_scores[opt_target] *= opt_dir
-        return df_scores
+        # NOTE: We use `.astype()` instead of `.apply(pd.to_numeric)`
+        # to catch incorrect values early.
```

Comment:
So, if this errors, it's basically a problem in the user script, right?

Can we handle this error in order to expose that comment directly to the user so they get some feedback on what to do?

For instance:

```python
try:
  ret = df_scores[list(self._opt_targets)].astype(float) * self._opt_targets.values()
except SomeExceptionThatPandasThrowsOnConversionErrors:
  _LOG.error("Failed to convert scores to floats. Double check parse script to ensure valid values are being provided: %s", json.dumps(the_scores)))
  raise e
```

Reply:
In this particular case, the problem is not just in the user script but also in our code that allowed such data to be stored in the database. I will create another PR with more checks there

Reply:
Since that's been pretty common, and it's not always clear what to do on a random exception, whether it's user based or internal issue, I think still some warning that describes at least "please ensure the scores are valid floats" would be a good idea.

Reply:
A comment in the code that explains that it might be due to bad values being mistakenly stored in the DB could also be helpful.

Reply:
done. there are unit tests, too